src/rules/index.ts

```diff
@@ -1,19 +1,29 @@
-import type Rule from './rule';
+import type RuleType from './rule';
 
 import {operator} from '../core/operators';
 import GroupRule from './group';
 import InverseRule from './inverse';
 import parse from './parse';
 
 export const rule = {
-  every<TContext>(rules: Array<Rule<TContext>>): Rule<TContext> {
+  every<TContext>(rules: Array<RuleType<TContext>>): RuleType<TContext> {
     return new GroupRule(operator.$and, rules);
   },
-  none<TContext>(rules: Array<Rule<TContext>>): Rule<TContext> {
+  none<TContext>(rules: Array<RuleType<TContext>>): RuleType<TContext> {
     return new InverseRule(new GroupRule(operator.$or, rules));
   },
   parse,
-  some<TContext>(rules: Array<Rule<TContext>>): Rule<TContext> {
+  some<TContext>(rules: Array<RuleType<TContext>>): RuleType<TContext> {
     return new GroupRule(operator.$or, rules);
   },
 };
+
+/**
+ * Allows you to define complex conditions and criteria for decision-making. It
+ * consists of one or more signals, which can be combined using logical
+ * operators to create intricate structures.
+ * 
+ * Takes a TContext argument which encapsulates the necessary information
+ * required by signals to make decisions and determine the outcome of rules.
+ */
```

Comment:
What do you think of moving this comment to the `Rule` class? That way we wouldn't need to define the internal `RuleType`. And the type can be re-exported with `export type {default as Rule} from './rule';`

Reply:
Great idea! Addressed in [1d6ecac](https://github.com/decs/ruls/pull/2/commits/1d6ecac4b6ac81e5628c6ea1bc4cec1f3c672601) 🙂